from0k2bp.tex

```diff
@@ -1080,73 +1093,80 @@
 and we'll therefore need to provide commitments in advance for each of
 these three coefficients. However, we already have the coefficient of $e^2$;
 that's $C_z$, which was given in advance. So we'll therefore need to provide
-\emph{two} additional commitments:
-\begin{align*}
-& C_1 = t_1 H + \left(\mathbf{x} \cdot \mathbf{d}_y + \mathbf{y} \cdot \mathbf{d}_x\right)G \\
-& C_0 = t_0 H + \left(\mathbf{d}_x \cdot \mathbf{d}_y \right)G \\
-\end{align*}
-(Note the use of $G$ not $\mathbf{G}$ because dot products are scalars, not
-vectors).
+\emph{two} additional commitments\footnote{Note the use of $G$ not $\mathbf{G}$ because dot products are scalars, not vectors.}:
+\begin{align}
+C_1 &:= t_1 H + \left(\mathbf{x} \cdot \mathbf{d}_y + \mathbf{y} \cdot \mathbf{d}_x\right)G \label{inner_product_proof_C0C1_def:0} \\
+C_0 &:= t_0 H + \left(\mathbf{d}_x \cdot \mathbf{d}_y \right)G  \label{inner_product_proof_C0C1_def:1}
+\end{align}
+
+So to do all this in the commitment step, the Prover had to come up with:
 
-So to do all this in the commitment step, the Prover had to come up with
-4 random scalars $r_d, s_d, t_1, t_0$ and two random vectors $\mathbf{d}_x, \mathbf{d}_y$ and then send 4 Pedersen
-commitments using this data: $A_d, B_d, C_1, C_0$.
+\begin{itemize}
+\item 4 random scalars $r_d, s_d, t_1, t_0$,
+\item 2 random vectors $\mathbf{d}_x, \mathbf{d}_y$, and 
+\item 4 Pedersen commitments using this data: $A_d, B_d, C_1, C_0$.
+\end{itemize}
 
 \hypertarget{the-challenge-step}{%
 \subsection[The challenge
 step]{\texorpdfstring{\protect\hypertarget{anchor-37}{}{}The challenge
 step}{The challenge step}}\label{the-challenge-step}}
 
-Nothing to discuss here -- the Verifier simply sends a single scalar
-value $e$.
+The Verifier just simply sends a single scalar value $e$ and the challenge step is finished.
 
 \hypertarget{the-response-step}{%
 \subsection[The response
 step]{\texorpdfstring{\protect\hypertarget{anchor-38}{}{}The response
 step}{The response step}}\label{the-response-step}}
 
 The above detailed discussion will hopefully make the following set of
-data, sent by the Prover, less bewildering:
-\begin{align*}
-& \mathbf{f}_x = e\mathbf{x} + \mathbf{d}_x \\
-& \mathbf{f}_y = e\mathbf{y} + \mathbf{d}_y \\
-& r_x = er + r_d \\
-& s_y = es + s_d \\
-& t_z = e^2 t + et_1 + t_0 \\
-\end{align*}
-
-First, note that here we are sending the blinded forms $\mathbf{f}_x, \mathbf{f}_y$ of the two
-vectors, not the Pedersen commitments -- the idea is that the Verifier
+data, sent by the Prover, less bewildering\footnote{One may forget what $t$~variable 
```

Comment:
Whilst this note is certainly not unhelpful, I would have to counter that - the reader may well forget what any number of other variables refers to, too :)

Reply:
Sure. It's just in this particular case, one need to go back 2 pages (if I remember correctly). That's why I have added the note.

Reply:
Yes it's a good idea.